pkg/service/model_build_target_group.go

```diff
@@ -0,0 +1,392 @@
+	if err != nil {
+		return nil, err
+	}
+	timeoutSeconds, err := t.buildTargetGroupHealthCheckTimeoutSeconds(ctx)
+	if err != nil {
+		return nil, err
+	}
+	healthyThresholdCount, err := t.buildTargetGroupHealthCheckHealthyThresholdCount(ctx)
+	if err != nil {
+		return nil, err
+	}
+	unhealthyThresholdCount, err := t.buildTargetGroupHealthCheckUnhealthyThresholdCount(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return &elbv2model.TargetGroupHealthCheckConfig{
+		Port:                    &healthCheckPort,
+		Protocol:                &healthCheckProtocol,
+		Path:                    healthCheckPathPtr,
+		IntervalSeconds:         &intervalSeconds,
+		TimeoutSeconds:          &timeoutSeconds,
+		HealthyThresholdCount:   &healthyThresholdCount,
+		UnhealthyThresholdCount: &unhealthyThresholdCount,
+	}, nil
+}
+
+var invalidTargetGroupNamePattern = regexp.MustCompile("[[:^alnum:]]")
+
+func (t *defaultModelBuildTask) buildTargetGroupName(_ context.Context, svcPort intstr.IntOrString, tgPort int64,
+	targetType elbv2model.TargetType, tgProtocol elbv2model.Protocol, hc *elbv2model.TargetGroupHealthCheckConfig) string {
+	healthCheckProtocol := string(elbv2model.ProtocolTCP)
+	healthCheckInterval := strconv.FormatInt(t.defaultHealthCheckInterval, 10)
+	if hc.Protocol != nil {
+		healthCheckProtocol = string(*hc.Protocol)
+	}
+	if hc.IntervalSeconds != nil {
+		healthCheckInterval = strconv.FormatInt(*hc.IntervalSeconds, 10)
+	}
+	uuidHash := sha256.New()
+	_, _ = uuidHash.Write([]byte(t.clusterName))
+	_, _ = uuidHash.Write([]byte(t.service.UID))
+	_, _ = uuidHash.Write([]byte(strconv.Itoa(int(tgPort))))
+	_, _ = uuidHash.Write([]byte(svcPort.String()))
+	_, _ = uuidHash.Write([]byte(targetType))
+	_, _ = uuidHash.Write([]byte(tgProtocol))
+	_, _ = uuidHash.Write([]byte(healthCheckProtocol))
+	_, _ = uuidHash.Write([]byte(healthCheckInterval))
+	uuid := hex.EncodeToString(uuidHash.Sum(nil))
+
+	sanitizedNamespace := invalidTargetGroupNamePattern.ReplaceAllString(t.service.Namespace, "")
+	sanitizedName := invalidTargetGroupNamePattern.ReplaceAllString(t.service.Name, "")
+	return fmt.Sprintf("k8s-%.8s-%.8s-%.10s", sanitizedNamespace, sanitizedName, uuid)
+}
+
+func (t *defaultModelBuildTask) buildTargetGroupAttributes(_ context.Context) ([]elbv2model.TargetGroupAttribute, error) {
+	var rawAttributes map[string]string
+	if _, err := t.annotationParser.ParseStringMapAnnotation(annotations.SvcLBSuffixTargetGroupAttributes, &rawAttributes, t.service.Annotations); err != nil {
+		return nil, err
+	}
+	if rawAttributes == nil {
+		rawAttributes = make(map[string]string)
+	}
+	if _, ok := rawAttributes[tgAttrsProxyProtocolV2Enabled]; !ok {
+		rawAttributes[tgAttrsProxyProtocolV2Enabled] = strconv.FormatBool(t.defaultProxyProtocolV2Enabled)
+	}
+	proxyV2Annotation := ""
+	exists := t.annotationParser.ParseStringAnnotation(annotations.SvcLBSuffixProxyProtocol, &proxyV2Annotation, t.service.Annotations)
```

Comment:
nit, better to be `if exists := ....; exists {}`.  do this will scope the exists variable inside the if statement.